app/core/util.py

```diff
@@ -73,3 +83,125 @@ def tree_table_to_json(
+    csv_document: DocumentValidationResult, db: Session
+) -> bool:
+    """
+    Compare the document provided in the csv against the document in the database to see if they are different. Then
+    update the database and S3 if they are different to represent the new data.
+    """
+    db_document = (
+        db.query(Document).filter(Document.import_id == csv_document.import_id).scalar()
+    )
+
+    if db_document is None:
+        # TODO how to handle this?
+        raise RuntimeError(
+            f"Could not find document with import_id {csv_document.import_id}"
+        )
+
+    geog_db_value = [
+        geo
+        for geo in db.query(Geography).filter(Geography.id == db_document.geography_id)
+    ][0].value
+
+    doc_type_db_value = [
+        doc_type
+        for doc_type in db.query(DocumentType).filter(
+            DocumentType.id == db_document.type_id
+        )
+    ][0].name
+
+    category_db_value = [
+        doc_cat
+        for doc_cat in db.query(Category).filter(Category.id == db_document.category_id)
+    ][0].name
+
+    update_results = DocumentUpdateResults(
+        name=UpdateResult(
+            csv_value=csv_document.create_request.name,
+            db_value=db_document.name,
+            updated=csv_document.create_request.name != db_document.name,
+        ),
+        publication_ts=UpdateResult(
+            csv_value=csv_document.create_request.publication_ts,
+            db_value=db_document.publication_ts,
+            updated=csv_document.create_request.publication_ts
+            != db_document.publication_ts,
+        ),
+        description=UpdateResult(
+            csv_value=csv_document.create_request.description,
+            db_value=db_document.description,
+            updated=csv_document.create_request.description != db_document.description,
+        ),
+        geography=UpdateResult(
+            csv_value=csv_document.create_request.geography,
+            db_value=geog_db_value,
+            updated=csv_document.create_request.geography != geog_db_value,
+        ),
+        type=UpdateResult(
+            csv_value=csv_document.create_request.type,
+            db_value=doc_type_db_value,
+            updated=csv_document.create_request.type != doc_type_db_value,
+        ),
+        category=UpdateResult(
+            csv_value=csv_document.create_request.category,
+            db_value=category_db_value,
+            updated=csv_document.create_request.category != category_db_value,
+        ),
+    )
+
+    _LOGGER.info(f"{update_results} for {csv_document.import_id}")
+    if any(
+        [
+            getattr(update_results, field.name).updated
+            for field in fields(DocumentUpdateResults)
+        ]
+    ):
+        update_doc_in_db(
+            updates=update_results, import_id=csv_document.import_id, db=db
+        )
+
+        delete_doc_in_s3(csv_document.import_id, PIPELINE_BUCKET, S3_PREFIXES)
+
+        return True
+
+    return False
```

Comment:
If we return `False` do we need to log a warning? Or info if the row is identical?